File: src/main/java/gargoyle/rpycg/util/Classes.java
package gargoyle.rpycg.util;

import javafx.collections.ObservableList;
import javafx.css.Styleable;
import org.jetbrains.annotations.NotNull;

public final class Classes {
    private Classes() {
        throw new IllegalStateException(Classes.class.getName());
    }

    public static void classAddRemove(@NotNull Styleable cell,
                                      @NotNull String classNameAdd, @NotNull String classNameRemove) {
        classAdd(cell, classNameAdd);
        classRemove(cell, classNameRemove);
    }

    public static void classAdd(@NotNull Styleable cell, @NotNull String className) {
        ObservableList<String> styleClass = cell.getStyleClass();
        if (!styleClass.contains(className)) {
            styleClass.add(className);
        }
    }

    @SuppressWarnings("MethodCallInLoopCondition")
    public static void classRemove(@NotNull Styleable cell, @NotNull String className) {
        ObservableList<String> styleClass = cell.getStyleClass();
        while (styleClass.contains(className)) {
            styleClass.remove(className);
        }
    }

    @SuppressWarnings("SameParameterValue")
    public static void classAddRemoveAll(@NotNull Styleable cell,
                                         @NotNull String classNameAdd, @NotNull String... classNameRemove) {
        classAdd(cell, classNameAdd);
        classRemoveAll(cell, classNameRemove);
    }

    public static void classRemoveAll(@NotNull Styleable cell, @NotNull String... classNames) {
        for (String className : classNames) {
            classRemove(cell, className);
        }
    }
}

File: app/src/main/kotlin/gargoyle/rpycg/util/Classes.kt
package gargoyle.rpycg.util

import javafx.css.Styleable

object Classes {

    fun classAdd(cell: Styleable, className: String) {
        if (!cell.styleClass.contains(className)) cell.styleClass.add(className)
    }

    fun classAddRemove(cell: Styleable, classNameAdd: String, classNameRemove: String) {
        if (!cell.styleClass.contains(classNameAdd)) cell.styleClass.add(classNameAdd)
        while (cell.styleClass.contains(classNameRemove)) cell.styleClass.remove(classNameRemove)
    }

    fun classAddRemoveAll(cell: Styleable, classNameAdd: String, vararg classNameRemove: String) {
        if (!cell.styleClass.contains(classNameAdd)) cell.styleClass.add(classNameAdd)
        for (className in classNameRemove) while (cell.styleClass.contains(className)) cell.styleClass.remove(className)
    }

    fun classRemove(cell: Styleable, className: String) {
        while (cell.styleClass.contains(className)) cell.styleClass.remove(className)
    }

    fun classRemoveAll(cell: Styleable, vararg classNames: String) {
        for (className in classNames) while (cell.styleClass.contains(className)) cell.styleClass.remove(className)
    }

}
